libs/Microsoft.MixedReality.WebRTC.Native/src/peer_connection.cpp

```diff
@@ -1274,63 +1670,620 @@ void PeerConnectionImpl::OnLocalDescCreated(
+  RTC_CHECK(mline_index >= 0);
+  rtc::CritScope lock(&tracks_mutex_);
+  if ((size_t)mline_index >= transceivers_.size()) {
+    // Insert empty entries for now; they should be filled when processing
+    // other added remote tracks or when finishing the transceiver status
+    // update.
+    while ((size_t)mline_index >= transceivers_.size() + 1) {
+      transceivers_.push_back(nullptr);
+    }
+    transceivers_.push_back(transceiver);
+  } else {
+    if (transceivers_[mline_index]) {
+      RTC_LOG(LS_ERROR) << "Trying to insert transceiver (name="
+                        << transceiver->GetName().c_str()
+                        << ") at mline index #" << mline_index
+                        << ", but another transceiver (name="
+                        << transceivers_[mline_index]->GetName().c_str()
+                        << ") already exists with the same index.";
+      return Error(Result::kUnknownError,
+                   "Duplicate transceiver for mline index");
+    }
+    transceivers_[mline_index] = transceiver;
+  }
+  return Error(Result::kSuccess);
+}
+
+ErrorOr<RefPtr<Transceiver>>
+PeerConnectionImpl::GetOrCreateTransceiverForSender(
+    mrsMediaKind media_kind,
+    webrtc::RtpSenderInterface* sender,
+    mrsTransceiverInteropHandle transceiver_interop_handle) {
+  RTC_DCHECK(MediaKindFromRtc(sender->media_type()) == media_kind);
+
+  // Find existing transceiver for sender
+  auto it = std::find_if(transceivers_.begin(), transceivers_.end(),
+                         [sender](const RefPtr<Transceiver>& tr) {
+                           return (tr && (tr->impl()->sender() == sender));
+                         });
+  if (it != transceivers_.end()) {
+    RTC_DCHECK(media_kind == it->get()->GetMediaKind());
+    RefPtr<Transceiver> transceiver = static_cast<Transceiver*>(it->get());
+    RTC_DCHECK_EQ(transceiver_interop_handle, transceiver->GetInteropHandle());
+    return transceiver;
+  }
+
+  std::string name = ExtractTransceiverNameFromSender(sender);
+
+  // Create new transceiver for local track
+  RefPtr<Transceiver> wrapper;
+  int mline_index = -1;
+  if (IsUnifiedPlan()) {
+    // Find transceiver implementation. It doesn't seem like there is a direct
+    // back-link, so iterate over all the peer connection transceivers.
+    auto transceivers = peer_->GetTransceivers();
+    auto it_tr = std::find_if(
+        transceivers.begin(), transceivers.end(),
+        [sender](auto const& tr) { return (tr->sender() == sender); });
+    if (it_tr == transceivers.end()) {
+      return Error(Result::kInvalidOperation,
+                   "Cannot match RTP sender with RTP transceiver.");
+    }
+    rtc::scoped_refptr<webrtc::RtpTransceiverInterface> impl = *it_tr;
+    mline_index = (int)std::distance(transceivers.begin(), it_tr);
+
+    // Create the transceiver wrapper
+    mrsTransceiverInitConfig config{};
+    config.desired_direction = Transceiver::Direction::kSendRecv;  //?
```

Comment:
//?

Reply:
Actually ended up being unused, so deleted. Current code reads that from the user if manually added, or defaults to `kRecvOnly` when added for a newly receiving track, which is what `libwebrtc` does internally in Unified Plan.

Fixed.